docs/connector-development/config-based/tutorial/3-connecting.md

```diff
@@ -0,0 +1,186 @@
+   This can be set in the requester definition.
+
+```
+requester:
+  type: HttpRequester
+  name: "{{ options['name'] }}"
+  url_base: "https://api.exchangeratesapi.io/v1/"
+```
+
+3. We can fetch the latest data by submitting a request to "/latest". This path is specific to the stream, so we'll set within the `rates_stream` definition.
+
+```
+rates_stream:
+  type: DeclarativeStream
+  options:
+    name: "rates"
+  primary_key: "id"
+  schema_loader:
+    ref: "*ref(schema_loader)"
+  retriever:
+    ref: "*ref(retriever)"
+    requester:
+      ref: "*ref(requester)"
+      path: "/latest"
+```
+
+4. Next, we'll set up the authentication.
+   The Exchange Rates API requires an access key, which we'll need to make accessible to our connector.
+   We'll configure the connector to use this access key by setting the access key in a request parameter and pointing to a field in the config, which we'll populate in the next step:
+
+```
+requester:
+  type: HttpRequester
+  name: "{{ options['name'] }}"
+  url_base: "https://api.exchangeratesapi.io/v1/"
+  http_method: "GET"
+  request_options_provider:
+    request_parameters:
+      access_key: "{{ config.access_key }}"
+```
+
+5. According to the ExchangeRatesApi documentation, we can specify the base currency of interest in a request parameter:
+
+```
+request_options_provider:
+  request_parameters:
+    access_key: "{{ config.access_key }}"
+    base: "{{ config.base }}"
+```
+
+6. Let's populate the config so the connector can access the access key and base currency.
+   First, we'll add these properties to the connector spec in
+   `source-exchange-rates-tutorial/source_exchange_rates_tutorial/spec.yaml`
+
+```
+documentationUrl: https://docs.airbyte.io/integrations/sources/exchangeratesapi
+connectionSpecification:
+  $schema: http://json-schema.org/draft-07/schema#
+  title: exchangeratesapi.io Source Spec
+  type: object
+  required:
+    - access_key
+    - base
+  additionalProperties: true
+  properties:
+    access_key:
+      type: string
+      description: >-
+        Your API Access Key. See <a
+        href="https://exchangeratesapi.io/documentation/">here</a>. The key is
+        case sensitive.
+      airbyte_secret: true
+    base:
+      type: string
+      description: >-
+        ISO reference currency. See <a
+        href="https://www.ecb.europa.eu/stats/policy_and_exchange_rates/euro_reference_exchange_rates/html/index.en.html">here</a>.
+      examples:
+        - EUR
+        - USD
+```
+
+7. We also need to fill in the connection config in the `secrets/config.json`
+   Because of the sensitive nature of the access key, we recommend storing this config in the `secrets` directory because it is ignored by git.
+
+```
+echo '{"access_key": "<your_access_key>", "base": "USD"}'  > secrets/config.json
+```
+
+We can now run the `check` operation, which verifies the connector can connect to the API source.
+
+```
+python main.py check --config secrets/config.json
```

Comment:
Running check fails for me: 
```
{"type": "CONNECTION_STATUS", "connectionStatus": {"status": "FAILED", "message": "Exception(\"failed to create object of type <class 'airbyte_cdk.sources.declarative.declarative_stream.DeclarativeStream'> because __init__() missing 1 required positional argument: 'name'\")"}}
```

Might be related to keeping the `$` . If i remove the `$` in my config I have: 
```
{"type": "CONNECTION_STATUS", "connectionStatus": {"status": "FAILED", "message": "Exception(\"failed to create object of type <class 'airbyte_cdk.sources.declarative.schema.json_schema.JsonSchema'> because __init__() missing 1 required positional argument: 'file_path'\")"}}
```

Here's my config for reference
[exchange_rates_tutorial.yaml.txt](https://github.com/airbytehq/airbyte/files/9243406/exchange_rates_tutorial.yaml.txt)

Reply:
😿 
added full sample here: https://github.com/airbytehq/airbyte/pull/15027/commits/ecf9b34f3759f68fa6bd54d61038c0363e6bcf55

fixed template: https://github.com/airbytehq/airbyte/pull/15209